ee/clickhouse/queries/cohort_query.py

```diff
@@ -0,0 +1,423 @@
+    if value is None:
+        raise ValueError(f"{value_name} cannot be None")
+    try:
+        parsed_value = int(value)
+    except ValueError:
+        raise ValueError(f"{value_name} must be an integer, got {value}")
+    if parsed_value < 0:
+        raise ValueError(f"{value_name} must be greater than 0, got {value}")
+    return parsed_value
+
+
+def relative_date_is_greater(date_1: Relative_Date, date_2: Relative_Date) -> bool:
+    if date_1[1] == date_2[1]:
+        return date_1[0] > date_2[0]
+
+    return relative_date_to_seconds(date_1) > relative_date_to_seconds(date_2)
+
+
+class CohortQuery(EnterpriseEventQuery):
+
+    BEHAVIOR_QUERY_ALIAS = "behavior_query"
+    _fields: List[str]
+    _events: List[str]
+
+    def __init__(
+        self,
+        filter: Filter,
+        team: Team,
+        round_interval=False,
+        should_join_distinct_ids=False,
+        should_join_persons=False,
+        # Extra events/person table columns to fetch since parent query needs them
+        extra_fields: List[ColumnName] = [],
+        extra_event_properties: List[PropertyName] = [],
+        extra_person_fields: List[ColumnName] = [],
+        override_aggregate_users_by_distinct_id: Optional[bool] = None,
+        **kwargs,
+    ) -> None:
+        self._fields = []
+        self._events = []
+        super().__init__(
+            filter=filter,
+            team=team,
+            round_interval=round_interval,
+            should_join_distinct_ids=should_join_distinct_ids,
+            should_join_persons=should_join_persons,
+            extra_fields=extra_fields,
+            extra_event_properties=extra_event_properties,
+            extra_person_fields=extra_person_fields,
+            override_aggregate_users_by_distinct_id=override_aggregate_users_by_distinct_id,
+            **kwargs,
+        )
+
+        property_groups = self._column_optimizer.property_optimizer.parse_property_groups(self._filter.property_groups)
+        self._inner_property_groups = property_groups.inner
+        self._outer_property_groups = property_groups.outer
+
+    def get_query(self) -> Tuple[str, Dict[str, Any]]:
+
+        if not self._outer_property_groups:
+            # everything is pushed down, no behavioural stuff to do
+            # thus, use personQuery directly
+            return self._person_query.get_query()
+
+        # TODO: clean up this kludge. Right now, get_conditions has to run first so that _fields is populated for _get_behavioral_subquery()
+        conditions, condition_params = self._get_conditions()
+        self.params.update(condition_params)
+
+        behavior_subquery, behavior_subquery_params = self._get_behavior_subquery()
+        self.params.update(behavior_subquery_params)
+
+        person_query, person_params = self._get_person_query()
+        self.params.update(person_params)
+
+        final_query = f"""
+        SELECT person_id FROM
```

Comment:
Most upstream stuff appears to assume this is `id`

```suggestion
        SELECT person_id as id FROM
```